### PR TITLE
Adult Buddy Fixes

### PR DESCRIPTION
**Pull request by randomPoison**
- Fixes the issue where adult buddies would reappear after midnight.
- Fixes the issue the stats gained from an adult buddy would go away after midnight.
- Fixes a bug in SoundManager that would cause errors if an audio source is inadvertently destroyed.
- Adds a separate spawn point for the player so that the player reappears in the garden each night rather than where they started the game.
- Adds a debug key for starting the midnight transition, currently defaults to "M".
- Makes the global shaders disable preview when the game stops running.

### CODE REVIEW

File: Assets/Scripts/Buddy/BuddyStats.cs [hunk 1]
```diff
@@ -74,6 +74,13 @@ enum HappinessState
 		Happy,
 	}
 
+	enum BuddyState
+	{
+		Normal,
+		Dead,
+		Adult,
```
Comment (brickman1444) [inline Assets/Scripts/Buddy/BuddyStats.cs:81]: Consider adding an invalid state as the first enum entry. An uninitialized enum defaults to the first value.

File: Assets/Scripts/Managers/SoundManager.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ AudioSource GetSource()
 
 		foreach ( AudioSource audioSource in audioObjs )
 		{
-			if ( !audioSource.isPlaying )
+			if ( audioSource && !audioSource.isPlaying )
```
Comment (brickman1444) [inline Assets/Scripts/Managers/SoundManager.cs]: A null reference should probably be removed from the list altogether.

**Comment by brickman1444** [comment]
Done reviewing.

File: Assets/Scripts/Buddy/BuddyStats.cs (hunk 1 above)
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyStats.cs:81]: I don't think that's needed in this case, I explicitly initialize the buddies to `Normal`.

File: Assets/Scripts/Managers/SoundManager.cs (hunk 2 above)
Comment (randomPoison) [inline Assets/Scripts/Managers/SoundManager.cs]: Done.

**Pull request merged**